src/App.tsx

```diff
@@ -15,35 +16,101 @@
   'Garlic',
 ];
 
+enum SortType {
+  DEFAULT,
+  ALPHABETICAL,
+  LENGTH,
+}
+
+type SortParams = {
+  sortType: SortType;
+  isReversed: boolean;
+};
+
+const getSortedGoods = (
+  items: string[],
+  { sortType, isReversed }: SortParams,
+) => {
+  const sortedList = [...goodsFromServer];
+
+  switch (sortType) {
+    case SortType.ALPHABETICAL:
+      sortedList.sort((a, b) => a.localeCompare(b));
+      break;
+    case SortType.LENGTH:
+      sortedList.sort((a, b) => a.length - b.length);
+      break;
+    default:
+      break;
+  }
+
+  return isReversed ? sortedList.reverse() : sortedList;
+};
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState<SortType>(SortType.DEFAULT);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const handleReset = () => {
+    setSortType(SortType.DEFAULT);
+    setIsReversed(false);
+  };
+
+  const sortedGoods = getSortedGoods(goodsFromServer, { sortType, isReversed });
+
+  const showResetButton = sortType !== SortType.DEFAULT || isReversed;
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortType !== SortType.ALPHABETICAL,
+          })}
+          onClick={() => setSortType(SortType.ALPHABETICAL)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortType !== SortType.LENGTH,
+          })}
+          onClick={() => setSortType(SortType.LENGTH)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={() => setIsReversed(current => !current)}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {showResetButton && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={handleReset}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
         <ul>
```

Comment:
[UX] - There is no indication of which sort option or direction is currently active. Consider adding a visual cue, such as a class or style, to the active button to communicate this to the user.